app/views/handlers/list_action.py

```diff
@@ -47,8 +47,22 @@ def handle_post(self):
             self.parent_location.block_id
         )
         self.questionnaire_store_updater.remove_answers(answer_ids_to_remove)
+        self.evaluate_and_update_section_status()
         self.questionnaire_store_updater.save()
 
+    def evaluate_and_update_section_status(self):
+        sections_to_evaluate = (
+            self.questionnaire_store_updater.get_in_progress_and_completed_sections()
```

Comment:
This currently seems to be returning all sections that are in progress and complete, and not just the ones that have a dependency on the list we are dealing with.